Source/Patches.cs

```diff
@@ -542,4 +1044,117 @@ public static void Postfix()
 			}
 		}
 	}
-}
\ No newline at end of file
+
+	[HarmonyPatch(typeof(UIRoot), nameof(UIRoot.UIRootUpdate))]
+	public static class UIRoot_UIRootUpdate_OpinionReport_Patch
+	{
+		static int opinionTicks = -1;
+		private static Dictionary<(Pawn, Pawn), int> lastOpinions = new Dictionary<(Pawn, Pawn), int>();
+
+		public static void Postfix()
+		{
+			try
+			{
+				var map = Find.CurrentMap;
+				if (map == null) return;
+
+				opinionTicks++;
+
+				if (opinionTicks % 12000 == 0)
```

Comment:
const to top of the class and maybe call all different code in UIRootUpdate on different ticks (or use a more advanced technique to spread load as discussed in earlier comments).